Not yet applied.  Needed on glibc-2_5-branch and HEAD.

2006-06-13  Thomas Schwinge  <tschwinge@gnu.org>

	* sysdeps/mach/hurd/tls.h: Change `ASSEMBLER' conditional to
	`__ASSEMBLER__'.
	* sysdeps/mach/hurd/i386/tls.h: Likewise.

Index: sysdeps/mach/hurd/tls.h
===================================================================
--- sysdeps/mach/hurd/tls.h.orig
+++ sysdeps/mach/hurd/tls.h
@@ -20,7 +20,7 @@
 #ifndef _TLS_H
 #define _TLS_H
 
-#if defined HAVE_TLS_SUPPORT && !defined ASSEMBLER
+#if defined HAVE_TLS_SUPPORT && !defined __ASSEMBLER__
 
 # include <stddef.h>
 # include <stdbool.h>
Index: sysdeps/mach/hurd/i386/tls.h
===================================================================
--- sysdeps/mach/hurd/i386/tls.h.orig
+++ sysdeps/mach/hurd/i386/tls.h
@@ -29,7 +29,7 @@
    thread pointer points to is unspecified.  Allocate the TCB there.  */
 # define TLS_TCB_AT_TP	1
 
-# ifndef ASSEMBLER
+# ifndef __ASSEMBLER__
 
 /* Use i386-specific RPCs to arrange that %gs segment register prefix
    addresses the TCB in each thread.  */
@@ -165,7 +165,7 @@
   return err;
 }
 
-# endif	/* !ASSEMBLER */
+# endif	/* !__ASSEMBLER__ */
 #endif /* HAVE_TLS_SUPPORT */
 
 #endif	/* i386/tls.h */
Index: argp/argp-help.c
===================================================================
--- argp/argp-help.c.orig
+++ argp/argp-help.c
@@ -42,6 +42,7 @@
 # endif
 #endif
 
+#include <stdbool.h>
 #include <stddef.h>
 #include <stdlib.h>
 #include <string.h>
Index: posix/wordexp.c
===================================================================
--- posix/wordexp.c.orig
+++ posix/wordexp.c
@@ -28,6 +28,7 @@
 #include <paths.h>
 #include <pwd.h>
 #include <signal.h>
+#include <stdbool.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
Index: stdlib/fmtmsg.c
===================================================================
--- stdlib/fmtmsg.c.orig
+++ stdlib/fmtmsg.c
@@ -19,6 +19,7 @@
 
 #include <fmtmsg.h>
 #include <bits/libc-lock.h>
+#include <stdint.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
